python/pyspark/sql/types.py

```diff
@@ -188,6 +199,8 @@ class IntegerType(PrimitiveType):
 
     The data type representing int values.
     """
+    def simpleString(self):
+        return 'int'
```

Comment:
There are dumped as JSON, passed into JVM, parsed again.

The good part is that we could infer the schema in pure Python (also in workers, it's necessary to infer the schema on   many rows of the RDD), the bad part is that we have duplicated code in Python and Scala.